File: packages/core-mobile/android/app/src/main/java/com/avaxwallet/SecureActivity.java
package com.avaxwallet;

import android.view.WindowManager;

import androidx.annotation.NonNull;

import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;

public class SecureActivity extends ReactContextBaseJavaModule {
    SecureActivity(ReactApplicationContext context) {
        super(context);
    }

    @ReactMethod
    public void onCreate() {
        getCurrentActivity().runOnUiThread(() -> {
            getCurrentActivity().getWindow().setFlags(WindowManager.LayoutParams.FLAG_SECURE, WindowManager.LayoutParams.FLAG_SECURE);
        });

    }

    @ReactMethod
    public void onDestroy() {
        getCurrentActivity().runOnUiThread(() -> {
            getCurrentActivity().getWindow().clearFlags(WindowManager.LayoutParams.FLAG_SECURE);
        });
    }

    @NonNull
    @Override
    public String getName() {
        return "SecureActivity";
    }
}

File: packages/core-mobile/android/app/src/main/java/com/avaxwallet/SecureActivity.kt
package com.avaxwallet

import android.view.WindowManager
import androidx.annotation.NonNull
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod

class SecureActivity(context: ReactApplicationContext) : ReactContextBaseJavaModule(context) {
    private val reactContext = context

    @ReactMethod
    fun onCreate() {
        val activity = reactContext.currentActivity ?: return
        activity.runOnUiThread {
            activity.window?.setFlags(
                WindowManager.LayoutParams.FLAG_SECURE, WindowManager.LayoutParams.FLAG_SECURE
            )
        }
    }

    @ReactMethod
    fun onDestroy() {
        val activity = reactContext.currentActivity ?: return
        activity.runOnUiThread {
            activity.window?.clearFlags(WindowManager.LayoutParams.FLAG_SECURE)
        }
    }

    @NonNull
    override fun getName(): String {
        return "SecureActivity"
    }
}
